File: 160-intersection-of-two-linked-lists/160-intersection-of-two-linked-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function(headA, headB) {
    let mp = new Map();
    let  p = headA, q = headB;
    while(p !== null){
        mp.set(p,1);
        p = p.next;
    }
    while(q !== null){
        if(mp.get(q) > 0)
            return q;
        q = q.next;
    }
    return q;
};

File: 160-intersection-of-two-linked-lists/160-intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        map<ListNode *,int> mp;
        ListNode * A = headA,*B = headB;
        while(A)
        {
            mp[A]++;
            A = A->next;
        }
        while(B)
        {
            if(mp[B] > 0)
                return B;
            B = B->next;
        }
        return NULL;
    }
};